src/Microsoft.Azure.WebJobs.Host/Lease/LeaseFactory.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Azure.WebJobs.Host.Lease;
+using Microsoft.Azure.WebJobs.Host.Storage;
+
+namespace Microsoft.Azure.WebJobs.Host.Lease
+{
+    /// <summary>
+    /// ILeaseProxy factory
+    /// </summary>
+    internal class LeaseFactory
+    {
+        public static ILeaseProxy CreateLeaseProxy(IStorageAccountProvider storageAccountProvider)
+        {
+            ILeaseProxy leaseProxy = null;
+
+            // Choose between Sql and Blob based lease implementations
+            if (SqlLeaseProxy.IsSqlLeaseType())
```

Comment:
>if (SqlLeaseProxy.IsSqlLeaseType()) [](start = 12, length = 35)

this will try to connect using SQL which is an expensive call just to decide the implementation type. The default connection timeout is 15 seconds. I think we should just parse the connection string to detect the implementation type